localScrapper.js

```diff
@@ -16,8 +27,15 @@ function getFileInfos(fileName, dirPath, stat) {
   let languageString
   let extension
 
-  const lastIndexOfSlash = dirPath.lastIndexOf('/')
-  const dirName = lastIndexOfSlash !== -1 ? dirPath.substr(lastIndexOfSlash + 1) : '.'
+  const directories = dirPath.split('/')
+  // Group by subdirectories : Take the name of the last directory (/karaoke/TEMP/Anime >> Anime)
+  // Group by top directory : Take the name of the topmost directory, excluding the base one :
+  //   /karaoke/TEMP/Anime >> TEMP
+  //   if anything is in the top directory, it will be indexed as such.
+  const dirName = groupBySubdirectories ?
+      directories[directories.length - 1] :
+      directories[1] || directories[0]
+
```

Comment:
Oh if we can use lodash, all the better! Event if I now prefer Ramda.

But I really don't like the syntax of a ternary coupled with an or statement :/ Don't we have something better?